TrafficLight.py

```diff
@@ -0,0 +1,111 @@
+            return False
+        # runs all checks on each traffic state matrix
+        for matrix in self.traffic_flow_matrix_list:
+            if not self.checkMatrixInvariants(matrix):
+                return False
+        # Checks no two names are the same
+        if len(set(self.name_list)) != len_rows:
+            return False
+        # checks to ensure the number of timers is equal to the number of states
+        if len(self.timers) == len_rows:
+            return False
+        else:
+            return True
+    """
+    Checks that the matrix involved follows all invariant rules
+    """
+    def checkMatrixInvariants(self, matrix):
+        # Checks all entries are floats [0, 1] and that every row sums to exactly 1
+        len_rows = len(self.name_list)
+        for row in matrix:
+            rowTotal = 0
+            for weight in row:
+                rowTotal += weight
+                if weight > 1 or weight < 0:
+                    return False
+            if rowTotal != 1:
+                return False
+        # Checks to ensure there is always 0 flow from a place to itself
+        for index, row in matrix:
+            if row[index] != 0:
+                return False
+        # Checks to make sure number of rows is same as number of columns
+        if len(matrix) != len_rows:
+            return False
+        # Checks to make sure that length of all rows is the same for all rows
+        for row in self.state_matrix:
+            if len(row) != len_rows:
+                return False
+        return True
+
+    """
+    :param from_name: the name of the place traffic is flowing from
+    :type string
+    :param to_name: the name of the place traffic is flowing to
+    :type string
+    :return A number between 0 and 1 representing the factor by which traffic
+     is scaled going in this direction
+    :rtype float
+    """
+
+    def get_traffic_flow(self, from_name, to_name):
+        rowIndex = self.name_list.index(from_name)
+        colIndex = self.name_list.index(to_name)
+        return self.traffic_matrix[rowIndex][colIndex]
+    """
+    Gets flow from [from_name] to [to_name]
+    """
+    def get_flow_in_current_state(self, from_name, to_name):
+        rowIndex = self.name_list.index(from_name)
+        colIndex = self.name_list.index(to_name)
+        return self.traffic_flow_matrix_list[self.current_state][rowIndex][colIndex]
+    """
+    Checks that the light has an entrance with the given name
+    """
+    def has_entrance(self, name):
+        return name in self.name_list
+
+    """
+    Adjusts to a new traffic state based on the amount of time that has elapsed.
+    :param time_elapsed the number of seconds that has elapsed since the last update
+    :type int
+    """
+
+    def change_state(self, time_elapsed):
+        totalTime = self.elapsed_time_in_state + time_elapsed
+        while totalTime > self.timers[current_state]:
+            totalTime -= self.timers[current_state]
```

Comment:
how is current_state defined?

this seems confusing, and I feel like it shouldn't work...